lib/MetaCPAN/Client.pm

```diff
@@ -200,10 +200,36 @@ sub all {
 }
 
 sub download_url {
-    my $self   = shift;
-    my $module = shift;
+    my $self             = shift;
+    my $module           = shift;
+    my $version_or_range = shift;
+    my $dev              = shift;
+
+    my $uri = $module;
+    my @extra;
+    if ( defined $version_or_range ) {
+
+        my @valid_ranges =  qw{ == != <= >= < > ! };
+        my $is_using_range;
+        foreach my $range ( @valid_ranges ) {
+            if ( index( $version_or_range, $range ) >= 0 ) {
```

Comment:
I agree with you `>1.01,<=2.00` is a weird request, but looks like this is what the API accept right now.
view: https://github.com/metacpan/metacpan-api/blob/master/t/server/controller/download_url.t#L53

What is the answer of that request, is not part metacpan-client side IMO. And it can change overtime, if it changes upstream.

I see you are saying when we have multiple rows to the answer for this request we are going to pick the first one.
Which is maybe or not sorted, but seems to be the lowest value.
To be honest I agree with you that picking the highest value is probably a better choice.
Could not we see this as an improvement? and a follow up fix later?

My main goal here is to be able to get the url for a specific version using `==` operator...
the range support is just a bonus for completeness, I guess when/if people start using it we can then improve it

Reply:
of course